clients/batchValidator/batchValidator.go

```diff
@@ -0,0 +1,104 @@
+package batchValidatorManagement
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-eth-bridge/clients"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+const minRequestTime = time.Millisecond
+const logPath = "BatchValidator"
+
+// ArgsBatchValidator is the DTO used for the creating a new batch validator instance
+type ArgsBatchValidator struct {
+	SourceChain      clients.Chain
+	DestinationChain clients.Chain
+	RequestURL       string
+	RequestTime      time.Duration
+}
+
+type batchValidator struct {
+	requestURL  string
+	requestTime time.Duration
+	log         logger.Logger
+	httpClient  HTTPClient
+}
+
+// NewBatchValidator returns a new batch validator instance
+func NewBatchValidator(args ArgsBatchValidator) (*batchValidator, error) {
+	err := checkArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	bv := &batchValidator{
+		requestURL:  args.RequestURL + "/" + string(args.SourceChain) + "/" + string(args.DestinationChain),
+		requestTime: args.RequestTime,
+		httpClient:  http.DefaultClient,
+	}
+	bv.log = logger.GetOrCreate(logPath)
+	return bv, nil
+}
+
+func checkArgs(args ArgsBatchValidator) error {
+	if args.RequestTime < minRequestTime {
+		return fmt.Errorf("%w in checkArgs for value RequestTime", clients.ErrInvalidValue)
+	}
+
+	return nil
+}
+
+func (bv *batchValidator) ValidateBatch(batch string) (bool, error) {
```

Comment:
could have injected the context here and pass it to bv.doRequest function

Reply:
added